src/client/commands/wait.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{rpc_call, JobId};
+
+use crate::client::job::job_status;
+use crate::client::status::{is_terminated, Status};
+use crate::transfer::connection::ClientConnection;
+use crate::transfer::messages::{FromClientMessage, JobInfoRequest, JobSelector, ToClientMessage};
+use std::thread::sleep;
+use std::time::Duration;
+
+pub async fn wait_on_job(
+    mut connection: ClientConnection,
+    selector: JobSelector,
+) -> anyhow::Result<()> {
+    let connection_ref = &mut connection;
+    let response = rpc_call!(
+        connection_ref,
+        FromClientMessage::JobInfo(JobInfoRequest {
+            selector,
+        }),
+        ToClientMessage::JobInfoResponse(r) => r
+    )
+    .await?;
+    let job_ids: Vec<JobId> = response
+        .jobs
+        .iter()
+        .filter(|job_info| {
+            vec![Status::Running, Status::Waiting]
+                .iter()
+                .any(|status| status == &job_status(&job_info))
+        })
+        .map(|info| info.id)
+        .collect();
+
+    if job_ids.is_empty() {
+        log::warn!("There are no jobs to wait for");
+    } else {
+        let mut current_jobs = 0;
+        let total_jobs = job_ids.len();
+
+        for job_id in job_ids {
+            let mut is_running = true;
+            while is_running {
+                let connection_ref = &mut connection;
+                let info_response = rpc_call!(
+                    connection_ref,
+                    FromClientMessage::JobInfo(JobInfoRequest {
+                        selector: JobSelector::Specific(vec![job_id]),
+                    }),
+                    ToClientMessage::JobInfoResponse(r) => r
+                )
+                .await?;
+
+                let info_opt = info_response.jobs.get(0);
+                if let Some(info) = info_opt {
+                    is_running = !is_terminated(info);
+                    let total = info.counters.n_running_tasks
+                        + info.counters.n_canceled_tasks
+                        + info.counters.n_failed_tasks
+                        + info.counters.n_finished_tasks;
+                    println!("{}: {} %", job_id, total * 100 / info.n_tasks);
+                    sleep(Duration::from_secs(1));
+                } else {
+                    is_running = false;
+                    log::error!("Job {} does not exist", job_id);
+                }
+            }
+            log::info!("Job {} has finished", job_id);
+
+            current_jobs += 1;
+            println!("Jobs done: {} / {}", current_jobs, total_jobs);
```

Comment:
It would be nice to display some kind of progress bar :) Could you try to take a look at https://docs.rs/indicatif/0.16.2/indicatif/?

Reply:
Maybe indicatif is too much, but the progress bar as you made in Job info would be fine

Reply:
That one was for a single use-case, I would rather not reinvent the wheel. But it's your call.